### PR TITLE
Create Aspect Models from AAS submodel template

### PR DESCRIPTION
**Pull request by Yauhenikapl**
Changes:
- Add new command for AAS submodels generation;
- Update AAS Generator.

Fixes: #468

### CODE REVIEW

File: tools/samm-cli/src/main/java/org/eclipse/esmf/aas/AasListSubmodelsCommand.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+   private LoggingMixin loggingMixin;
+
+   @CommandLine.ParentCommand
+   private AasCommand parentCommand;
+
+   @Override
+   public void run() {
+      final String path = parentCommand.getInput();
+      final String extension = FilenameUtils.getExtension( path );
+      if ( !extension.equals( "xml" ) && !extension.equals( "json" ) && !extension.equals( "aasx" ) ) {
```
Comment (atextor) [inline tools/samm-cli/src/main/java/org/eclipse/esmf/aas/AasListSubmodelsCommand.java:33]: Not necessary, but only a suggestion:

```suggestion
      if ( Arrays.stream( AasFileFormat.values() ).anyMatch( format -> format.toString().equals( extension ) ) ) {
```

File: tools/samm-cli/src/main/java/org/eclipse/esmf/aas/AasListSubmodelsCommand.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+         throw new CommandException( "Input file name must be an .xml, .aasx or .json file" );
```
Comment (atextor) [inline tools/samm-cli/src/main/java/org/eclipse/esmf/aas/AasListSubmodelsCommand.java:34]: Suggestion:

```suggestion
         throw new CommandException( "Input file name must be one of " + AasFileFormat.allValues() );
```

File: tools/samm-cli/src/main/java/org/eclipse/esmf/aas/AasListSubmodelsCommand.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      }
+
+      List<String> submodelNames = AasToAspectModelGenerator.fromFile( new File( path ) ).getSubmodelNames();
```
Comment (atextor) [inline tools/samm-cli/src/main/java/org/eclipse/esmf/aas/AasListSubmodelsCommand.java:37]:
```suggestion
      final List<String> submodelNames = AasToAspectModelGenerator.fromFile( new File( path ) ).getSubmodelNames();
```

File: tools/samm-cli/src/main/java/org/eclipse/esmf/aas/AasListSubmodelsCommand.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+      for ( String submodelName : submodelNames ) {
+         int count = submodelNames.indexOf( submodelName );
+         if ( count < 10 ) {
+            System.out.printf( " %s: %s%n", count + 1, submodelName );
+         } else {
+            System.out.printf( "%s: %s%n", count + 1, submodelName );
+         }
+      }
```
Comment (atextor) [inline tools/samm-cli/src/main/java/org/eclipse/esmf/aas/AasListSubmodelsCommand.java:46]:
```suggestion
      for ( final String submodelName : submodelNames ) {
         final int index = submodelNames.indexOf( submodelName );
         System.out.printf( "%2s: %s%n", index + 1, submodelName );
      }
```

File: tools/samm-cli/src/main/java/org/eclipse/esmf/aas/to/AasToAspectCommand.java [hunk 5]
```diff
@@ -38,6 +40,10 @@ public class AasToAspectCommand extends AbstractCommand {
    @CommandLine.Option( names = { "--output-directory", "-d" }, description = "Output directory to write files to" )
    private String outputPath = ".";
 
+   @CommandLine.Option( names = { "--submodel-template", "-s" },
+         description = "Select one or a few options for generation an Aspect Model from AAS. Before run 'list' command for getting possible options." )
```
Comment (atextor) [inline tools/samm-cli/src/main/java/org/eclipse/esmf/aas/to/AasToAspectCommand.java:44]:
```suggestion
         description = "Select the submodel template(s) to include, as returned by the aas list command" )
```

**Review by atextor — CHANGES_REQUESTED** [review]

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 6]
```diff
@@ -136,9 +136,14 @@ The available options and their meaning can also be seen in the help text of the
...
-.2+| aas <aas file> to aspect      | Translate Asset Administration Shell (AAS) Submodel Templates to Aspect Models          | `samm aas AssetAdminShell.aasx to aspect`
+.3+| aas <aas file> to aspect      | Translate Asset Administration Shell (AAS) Submodel Templates to Aspect Models          | `samm aas AssetAdminShell.aasx to aspect`
                                    | _--output-directory, -d_ : output directory to write files to (default:
                                        current directory)                                                                    |
+                                   | _--submodel-template, -s_ : selected submodel template for generating;
+                                       run `samm aas <aas file> list` to list them.                                          | `samm aas AssetAdminShell.aasx to aspect -s 1 -s 2`
+.1+| aas <aas file> list           | Retrieve a list of submodel templates contained within the provided
+                                       Asset Administration Shell (AAS) file.                                                | `samm ass AssetAdminShell.aasx list`
```
Comment (atextor) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:145]:
```suggestion
                                       Asset Administration Shell (AAS) file.                                                | `samm aas AssetAdminShell.aasx list`
```

File: tools/samm-cli/src/test/java/org/eclipse/esmf/ProcessLauncher.java [hunk 7]
```diff
@@ -48,9 +47,9 @@ public ExecutionResult runAndExpectSuccess( final String... arguments ) {
       return result;
    }
 
-   public static record ExecutionContext( List<String> arguments, Optional<byte[]> stdin, File workingDirectory ) {
+   public static record ExecutionContext(List<String> arguments, Optional<byte[]> stdin, File workingDirectory) {
    }
 
-   public static record ExecutionResult( int exitStatus, String stdout, String stderr, byte[] stdoutRaw, byte[] stderrRaw ) {
+   public static record ExecutionResult(int exitStatus, String stdout, String stderr, byte[] stdoutRaw, byte[] stderrRaw) {
```
Comment (atextor) [inline tools/samm-cli/src/test/java/org/eclipse/esmf/ProcessLauncher.java:53]: Undo these changes; it seems the formatting style is up to date. I'll have a look at https://github.com/eclipse-esmf/esmf-sdk/issues/25 for this.

**Review by atextor — CHANGES_REQUESTED** [review]

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc (hunk 6 above)
Comment (Yauhenikapl) [reply documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:145]: Oh my God... 
Sorry for this...

**Review by atextor — APPROVED** [review]

**Pull request merged**